api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -350,4 +355,127 @@ private OrderSet orderSetBuilder(boolean orderSetRetired, boolean orderSetMember
+	public void getOrderSetAttributeByUuid_shouldReturnNullIfNoOrderSetAttributeHasTheGivenUuid() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetService service = Context.getOrderSetService();
+		assertNull(service.getOrderSetAttributeByUuid("not-a-uuid"));
+	}
+
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByUuid(String)
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByUuid_shouldReturnTheOrderSetAttributeTypeWithTheGivenUuid() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		assertEquals("Audit Date", Context.getOrderSetService().getOrderSetAttributeTypeByUuid(
+		    "8516cc50-6f9f-33e0-8414-001e648eb67e").getName());
+	}
+
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByUuid(String)
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByUuid_shouldReturnNullIfNoOrderSetAttributeTypeExistsWithTheGivenUuid() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		assertNull(Context.getOrderSetService().getOrderSetAttributeTypeByUuid("not-a-uuid"));
+	}
+
+	/**
+	 * @see OrderSetService#purgeOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void purgeOrderSetAttributeType_shouldCompletelyRemoveAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(2, initialOrderSetAttributeTypesCount);
+		Context.getOrderSetService().purgeOrderSetAttributeType(Context.getOrderSetService().getOrderSetAttributeType(2));
+		assertEquals(initialOrderSetAttributeTypesCount - 1, Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+	}
+
+	/**
+	 * @see OrderSetService#retireOrderSetAttributeType(OrderSetAttributeType,String)
+	 */
+	@Test
+	public void retireOrderSetAttributeType_shouldRetireAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetAttributeType orderSetAttributeType = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertFalse(orderSetAttributeType.getRetired());
+		assertNull(orderSetAttributeType.getRetiredBy());
+		assertNull(orderSetAttributeType.getDateRetired());
+		assertNull(orderSetAttributeType.getRetireReason());
+		Context.getOrderSetService().retireOrderSetAttributeType(orderSetAttributeType, "for testing");
+		orderSetAttributeType = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertTrue(orderSetAttributeType.getRetired());
+		assertNotNull(orderSetAttributeType.getRetiredBy());
+		assertNotNull(orderSetAttributeType.getDateRetired());
+		assertEquals("for testing", orderSetAttributeType.getRetireReason());
+	}
+
+	/**
+	 * @see OrderSetService#saveOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void saveOrderSetAttributeType_shouldCreateANewOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(2, initialOrderSetAttributeTypesCount);
```

Comment:
What is the use of this assertion?

Reply:
@dkayiwa the use of this assertion is first to check and confirm the number of order set attribute types that we currently have in the dataset before we introduce a new one to make it 3

Reply:
Same as above.